### PR TITLE
Peribolos: update GH teams perms across org files

### PR DESCRIPTION
**Pull request by Priyankasaggu11929**
The PR 

- updates the following team repo permissions, to add members/bot accounts removed by Peribolos
(ref: https://kubernetes.slack.com/archives/CHGFYJVAN/p1690906660582369)
        - `sig-contributor-experience`
        - `stage-bot`
        - `release-team`
        - `sig-release`
        - `sig-testing`
        - `sig-k8s-infra`
        - `production-readiness`
        - `sig-cloud-provider`

- update `admin/update.sh` script to include `--skip-removals` flag

/sig contributor-experience
/assign @MadhavJivrajani 
/hold

### CODE REVIEW

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml [hunk 1]
```diff
@@ -346,18 +346,26 @@ teams:
...
+    - killianmuldoon
     # - kranurag7
     # - mjlshen
     - nawazkh
     - nprokopic
     - Prajyot-Parab
     # - razashahid107
+    - sbueringer
     # - Sunnatillo
+    - tobiasgiese
```
Comment (palnabarun) [inline config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:368]: This is a legit removal. Ref: https://github.com/kubernetes/org/pull/4353

**Comment by Priyankasaggu11929** [comment]
/hold

will fix the failing CI tests once done adding back all usernames.

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml (hunk 1 above)
Comment (palnabarun) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:368]: You can drop these changes since they are supposed to be removed.

Comment (CecileRobertMichon) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:368]: +1

File: config/kubernetes/sig-autoscaling/teams.yaml [hunk 2]
```diff
@@ -14,10 +14,15 @@ teams:
   autoscaler-maintainers:
```
Comment (palnabarun) [inline config/kubernetes/sig-autoscaling/teams.yaml:14]: The autoscaling cleanups are also legit. Ref: https://github.com/kubernetes/org/pull/4343

These can be safely dropped.

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml [hunk 3]
```diff
@@ -330,12 +332,17 @@ teams:
...
     - chrischdi
+    - figo
+    - frapposelli
     - gab-satchi
     - killianmuldoon
     - randomvariable
     - sbueringer
+    - sidharthsurana
     - srm09
+    - vincepri
```
Comment (palnabarun) [inline config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:345]: These can also be dropped. Ref: https://github.com/kubernetes/org/pull/4345

**Comment by palnabarun** [comment]
Keeping a living copy of the changes that can be dropped:
- config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml (#4345, #4345)
- config/kubernetes/sig-autoscaling/teams.yaml (#4343)

File: config/kubernetes/org.yaml [hunk 4]
```diff
@@ -2265,7 +2268,9 @@ teams:
       metrics: admin
       mount-utils: admin
       node-api: admin
-      noderesourcetopology-api: admin
+      # TODO: noderesourcetopology-api is part of k-sigs org
+      # noderesourcetopology-api: admin
```
Comment (palnabarun) [inline config/kubernetes/org.yaml:2272]: The noderesourcetopology-api repo was removed as a staging repo. We should be good to remove this repo assignment and the comment.

Ref: https://github.com/kubernetes/org/issues/4224#issuecomment-1618750306

File: config/kubernetes/sig-autoscaling/teams.yaml (hunk 2 above)
Comment (palnabarun) [reply config/kubernetes/sig-autoscaling/teams.yaml:14]: Note: Changes to the whole file can be dropped. GitHub UI is weird and doesn't show the whole selection.

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml (hunk 3 above)
Comment (palnabarun) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:345]: Note: Changes to the whole file can be dropped. GitHub UI is weird and doesn't show the whole selection.

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml (hunk 1 above)
Comment (palnabarun) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:375]: Note: Changes to the whole file can be dropped. GitHub UI is weird and doesn't show the whole selection.

**Review by MadhavJivrajani — CHANGES_REQUESTED** [review]
Dropping a note as a requested change to prevent accident merge: we need to include `--skip-removals` flag in `admin/update.sh`.

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml (hunk 1 above)
Comment (Priyankasaggu11929) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml]: dropped the respective commit. Thanks!

File: config/kubernetes/sig-autoscaling/teams.yaml (hunk 2 above)
Comment (Priyankasaggu11929) [reply config/kubernetes/sig-autoscaling/teams.yaml]: dropped the respective commit. Thanks!

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml (hunk 3 above)
Comment (Priyankasaggu11929) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:345]: dropped it as part of the respective entire commit. Thanks!

**Comment by Priyankasaggu11929** [comment]
@MadhavJivrajani, include the --skip-removals in the `admin/update.sh`.

From my side, the changes to add back members are done.

**Review by MadhavJivrajani — APPROVED** [review]
/lgtm
/approve
/hold cancel

File: config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml (hunk 3 above)
Comment (Priyankasaggu11929) [reply config/kubernetes-sigs/sig-cluster-lifecycle/teams.yaml:345]: Actually, just realised. I missed this. Will raise a PR to fix this tomorrow,

**Pull request merged**